src/integration_tests/tests/common-docker-idp.bash

```diff
@@ -0,0 +1,34 @@
+# Creates a clean environment where awscli-login is enabled and
+# configured.
+
+load 'base'
+eval "_base_$(declare -f setup)"  # Rename setup to _base_setup
+
+setup() {
+    _base_setup
```

Comment:
The following code should allow this PR to be merged:

```suggestion
    _base_setup

    # We cannot run integration tests dependent on Linux docker
    # containers on Windows runners because GitHub Actions does not
    # support Linux containers on Windows (See actions/runner-images#1143,
    # actions/runner#904, and actions/runner-images#5760).
    if [ $RUNNER_OS == "Windows" ]; then
        skip "Windows runners do not support Docker IdP integration tests."
    fi
    
```

Note that `_base_setup` has to happen before the skip, otherwise the teardown fails.

Reply:
I think it's fine to suppress these tests on Windows, since the challenges are test framework, not issues with the underlying tested tech. I don't think the OS matrix approach is critical to these tests.